### PR TITLE
completing the Activity delete

### PR DESCRIPTION
**Pull request by KavishanSukumar**

### CODE REVIEW

File: seaway.API/Controllers/ActivityController.cs [hunk 1]
```diff
@@ -146,5 +146,51 @@ public IActionResult PostActivity([FromForm]ActivityWithPicModel activity)
...
+                }
+                else
+                {
+                    return BadRequest("No Activity exist for this Id");
+                }
+
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError("An exception occurred while deleting room : " + ex.Message);
```
Comment (Nitharsha99) [inline seaway.API/Controllers/ActivityController.cs:190]: wrong log error message

**Comment by KavishanSukumar** [comment]
Comments resolved and completed SEA-20 Update function on Activity

File: seaway.API/Controllers/ActivityController.cs [hunk 2]
```diff
@@ -146,5 +146,95 @@ public IActionResult PostActivity([FromForm]ActivityWithPicModel activity)
...
+                _logger.LogError("An exception occurred while deleting Activity : " + ex.Message);
+                return BadRequest(ex.Message);
+            }
+        }
+
+
+        [HttpPut]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public IActionResult UpdateActivity([FromForm] Activity activity, int activityId)
```
Comment (Nitharsha99) [inline seaway.API/Controllers/ActivityController.cs:199]: Remove [FromForm]

File: seaway.API/Controllers/ActivityController.cs [hunk 3]
```diff
@@ -146,5 +146,95 @@ public IActionResult PostActivity([FromForm]ActivityWithPicModel activity)
...
+                    if (oldActivity.ActivityName != null)
+                    {
+                        Activity updateActivity = new Activity
+                        {
+                            ActivityName = activity.ActivityName,
+                            Description = activity.Description,
+                            IsActive = activity.IsActive,
+                 
+                        };
+                        _activityManager.UpdateActivity(updateActivity, activityId);
```
Comment (Nitharsha99) [inline seaway.API/Controllers/ActivityController.cs:215]: we need insert picDocuments of Activity if there is any pics

File: seaway.API/Manager/ActivityManager.cs [hunk 4]
```diff
@@ -181,5 +181,66 @@ public int PostActivity(Activity activity)
...
+            }
+        }
+
+        public void UpdateActivity(Activity activity, int activityId)
+        {
+            try
+            {
+                using (SqlConnection con = new SqlConnection(this._conString))
+                {
+                    using (SqlCommand cmd = new SqlCommand("[UpdateActivity]", con))
```
Comment (Nitharsha99) [inline seaway.API/Manager/ActivityManager.cs:219]: unwanted brackets

File: seaway.API/Scripts/25052024.sw.sql [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+	END
+END
+GO
+
+
+
+
+
+
+USE [seaway]
```
Comment (Nitharsha99) [inline seaway.API/Scripts/25052024.sw.sql:38]: remove this use [seaway] comment 
and put the if condition for checking the SP existings

**Comment by KavishanSukumar** [comment]
Second round comments resolved

**Pull request merged**